test/examples/ucp_client_server.c

```diff
@@ -332,6 +334,56 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
     return 0;
 }
 
+static char* sockaddr_get_ip_str(struct sockaddr_storage *sa,
+                                 char *ip, size_t max_size)
+{
+    struct sockaddr_in6 *addr_in6;
+    struct sockaddr_in *addr_in;
+    struct sockaddr sock_addr;
+
+    /* overcome older compiler versions' error:
+     * "error: dereferencing pointer ({anonymous}) does break strict-aliasing rules" */
+    memcpy(&sock_addr, sa, sizeof(sock_addr));
+
+    switch (sock_addr.sa_family) {
+    case AF_INET:
+        addr_in = (struct sockaddr_in *)&sock_addr;
+        inet_ntop(AF_INET, &addr_in->sin_addr, ip, max_size);
+        return ip;
+    case AF_INET6:
+        addr_in6 = (struct sockaddr_in6 *)&sock_addr;
+        inet_ntop(AF_INET6, &addr_in6->sin6_addr, ip, max_size);
+        return ip;
+    default:
+        return "Invalid address family";
+    }
+}
+
+static char* sockaddr_get_port_str(struct sockaddr_storage *sa,
+                                   char *port, size_t max_size)
+{
+    struct sockaddr_in6 addr_in6;
+    struct sockaddr_in addr_in;
+    struct sockaddr sock_addr;
+
+    /* overcome older compiler versions' error:
+     * "error: dereferencing pointer ({anonymous}) does break strict-aliasing rules" */
+    memcpy(&sock_addr, sa, sizeof(sock_addr));
```

Comment:
the internal variable sock_addr and this memcpy should go away, it's not really needed

Reply:
it's used to overcome this error "error: dereferencing pointer ({anonymous}) does break strict-aliasing rules" on an old compiler

Reply:
after adding addr_in6 and addr_in it's not needed any more
i see it's used only for sa_family, we can take that directly from input parameter